pytket/phir/phirgen.py

```diff
@@ -169,58 +169,81 @@ def classical_op(exp: LogicExp, *, bitwise: bool = False) -> JsonDict:
+    qop: JsonDict
+    match gate:
+        case "Measure":
+            qop = {
+                "qop": gate,
+                "returns": [arg_to_bit(cmd.bits[0])],
+                "args": [arg_to_bit(cmd.args[0])],
+            }
+        case ("CX"
+            | "CY"
+            | "CZ"
+            | "RXX"
+            | "RYY"
+            | "RZZ"
+            | "R2XXYYZZ"
+            | "SXX"
+            | "SXXdg"
+            | "SYY"
+            | "SYYdg"
+            | "SZZ"
+            | "SZZdg"
+            | "SWAP"
+        ):  # two-qubit gates  # fmt: skip
+            qop = {
+                "qop": gate,
+                "angles": angles,
+                "args": [[arg_to_bit(cmd.qubits[0]), arg_to_bit(cmd.qubits[1])]],
+            }
+        case _:  # single-qubit gates
+            qop = {
+                "qop": gate,
+                "angles": angles,
+                "args": [arg_to_bit(cmd.qubits[0])],
+            }
+    return qop
+
+
 def convert_subcmd(op: tk.Op, cmd: tk.Command) -> JsonDict | None:
-    """Return PHIR dict give op and its arguments."""
+    """Return PHIR dict given a tket op and its arguments."""
     if op.is_gate():
-        try:
-            gate = tket_gate_to_phir[op.type]
-        except KeyError:
-            if op.type == tk.OpType.Phase:
-                # ignore global phase
-                return {"mop": "Skip"}
-            logging.exception("Gate %s unsupported by PHIR", op.get_name())
-            raise
-        angles = (op.params, "pi") if op.params else None
-        qop: JsonDict
-        match gate:
-            case "Measure":
-                qop = {
-                    "qop": gate,
-                    "returns": [arg_to_bit(cmd.bits[0])],
-                    "args": [arg_to_bit(cmd.args[0])],
-                }
-            case ("CX"
-                | "CY"
-                | "CZ"
-                | "RXX"
-                | "RYY"
-                | "RZZ"
-                | "R2XXYYZZ"
-                | "SXX"
-                | "SXXdg"
-                | "SYY"
-                | "SYYdg"
-                | "SZZ"
-                | "SZZdg"
-                | "SWAP"
-            ):  # two-qubit gates  # fmt: skip
-                qop = {
-                    "qop": gate,
-                    "angles": angles,
-                    "args": [[arg_to_bit(cmd.qubits[0]), arg_to_bit(cmd.qubits[1])]],
-                }
-            case _:  # single-qubit gates
-                qop = {
-                    "qop": gate,
-                    "angles": angles,
-                    "args": [arg_to_bit(cmd.qubits[0])],
-                }
-        return qop
+        return convert_gate(op, cmd)
 
     out: JsonDict | None = None
     match op:  # non-quantum op
         case tk.BarrierOp():
-            out = {"meta": "barrier", "args": [arg_to_bit(qbit) for qbit in cmd.qubits]}
+            if op.data:
+                match op.data[0:5]:
```

Comment:
It is a little funny that this is specially is looking for 5 letter words... but looks like that happens to be the case so far, so works. lol

Reply:
Yeah, perhaps I should have added a comment there for where it comes from:

https://github.com/CQCL/tket/blob/0ec603986821d994caa3a0fb9c4640e5bc6c0a24/pytket/pytket/qasm/qasm.py#L419